smf/clickhouse/method_script.sh

```diff
@@ -28,13 +28,90 @@ ipadm show-addr "$DATALINK/ll" || ipadm create-addr -t -T addrconf "$DATALINK/ll
 ipadm show-addr "$DATALINK/omicron6"  || ipadm create-addr -t -T static -a "$LISTEN_ADDR" "$DATALINK/omicron6"
 route get -inet6 default -inet6 "$GATEWAY" || route add -inet6 default -inet6 "$GATEWAY"
 
-args=(
-  "--log-file" "/var/tmp/clickhouse-server.log"
-  "--errorlog-file" "/var/tmp/clickhouse-server.errlog"
-  "--"
-  "--path" "${DATASTORE}"
-  "--listen_host" "$LISTEN_ADDR"
-  "--http_port" "$LISTEN_PORT"
-)
-
-exec /opt/oxide/clickhouse/clickhouse server "${args[@]}" &
+# Retrieve addresses of the clickhouse nodes.
+CH_ADDRS="$(/opt/oxide/internal-dns-cli/bin/dnswait clickhouse -H \
```

Comment:
Does `dnswait` actually wait until there are _two_ addresses? Or until there are _any_ addresses? I'm wondering how we can avoid a cluster with 1 node, because `dnswait` returned only one record.

Reply:
Added a few checks to make sure we have the correct amount of hostnames with the correct format. Worth to note that the generation of these configs won't always be done via scripts https://github.com/oxidecomputer/omicron/issues/3824 .